### ISSUE TITLE
Additional eslint rules

### ISSUE THREAD

**Issue by TGCrystal** [OpenCircuits/OpenCircuits#1088]
Eslint is now set up for OpenCircuits, but there are still additional rules we would like to implement (or look at implementing). This issue will be used to collect all those into one place. Some are adding a rule, some are adding an entire plugin. Anything that seems like a variant of a currently existing rule can be opened as an issue on the respective linting repositories issue list,  As each is worked on, a separate issue can be created. This list can also be expanded as needed.
For writing custom rules, the guide https://sourcegraph.com/notebooks/Tm90ZWJvb2s6MTA2OA== looks like it may be useful.

- [ ] Check for deprecated function calls
  - There is a plugin called `eslint-plugin-deprecation` that does this and it is in use on the [eslint-deprecation branch](https://github.com/OpenCircuits/OpenCircuits/commit/69e0c7adcb1d1b4c83bd51b6b82c3d6c77a3f5be). There are seven warnings triggered, but duplicates aside there are 3 issues:
    1. `url.format` used three times in `scripts/webpack/index.ts`
    2. `createStore` from `redux`, used in the loading `index.ts` file for analog and digital
    3. `navigator.platform` used in `src/site/shared/utils/GetOS.ts`
- [ ] [eslint-plugin-functional](https://github.com/jonaskello/eslint-plugin-functional), particularly the `immutable-data` rule but others may also be worth checking out
- [ ] `indent` or more specifically, the typescript variant `@typescript-eslint/indent`. As of now (and probably a long time), it is officially considered broken (see https://github.com/typescript-eslint/typescript-eslint/issues/1824) but it may be possible to enable some parts of it. Will require some trial and error.
- [ ] `import/no-cycle` once the model refactor is done
- [ ] In `import/newline-after-import`, set the `considerComments` comments setting to true once the featue is released (releases: https://github.com/import-js/eslint-plugin-import/releases)
- [ ] Enable `jsdoc/require-jsdoc` if someone feels like adding jsdoc comments for literally everything in the project
- [ ] `jsx-a11y/no-noninteractive-element-interactions` (and possibly more `jsx-a11y` rules) to better improve accessibility and keyboard usage
- [ ] Consider setting the `noSortAlphabetically` property of `react/jsx-sort-props` to false
- [ ] A variation of `parens-new-line`, but only for non-fragment children. See the comment in `.react.cjs` for more details.
- [ ] Finalize `@typescript-eslint/naming-convention`
- [ ] Enable `unicorn/no-null` in `.tsx` files when https://github.com/microsoft/TypeScript/issues/21699 is resolved
- [ ] A variation of `@typescript-eslint/semi` that doesn't try to put in a semicolon directly following a `}`
- [ ] Parenthesis/multiline usage for ternary statements. See https://eslint.org/docs/latest/rules/multiline-ternary, https://github.com/getify/eslint-plugin-proper-ternary, and https://github.com/grayedfox/eslint-plugin-ternary.
- [ ] Parenthesis/braces in arrow functions. See https://eslint.org/docs/latest/rules/arrow-parens, https://eslint.org/docs/latest/rules/arrow-body-style, https://eslint.org/docs/latest/rules/arrow-spacing, and maybe https://eslint.org/docs/latest/rules/no-extra-parens
- [ ] Spaces before keywords in top level structures (like before `export` or `function`). Seems obvious but can't seem to find a rule. `align-import/trim-import` does this for `import` (we already have it enabled).
- [ ] Ban empty newlines between end of jsdoc comment and what the comment actually goes with.
- [ ] Space between imports (should be `import {V, Vector} from "Vector";` not `import {V,Vector} from "Vector";`)
- [ ] Ban multiple newlines between import groups
- [ ] Ban explicitly declaring `void` return type (https://typescript-eslint.io/rules/explicit-function-return-type/ does the opposite of this)
